### ISSUE TITLE
Update real time feeds to make use of dynamic entities custom data source

### ISSUE THREAD

**Issue by ldanzinger** [Esri/dynamic-situational-awareness-qt#367]
# Summary

DSA simulates real time data feeds with a message simulator that sends GeoMessage and Cursor On Target messages. The DSA app listens to a UDP port and displays the messages as graphics on the screen. If the message is an update to an existing graphic, the graphic attributes and geometry will be updated.

DSA was written before Runtime/Maps SDK had official support for real time feeds. Since then, we've added dynamic entities, which allow you to hook into stream services, as well as custom data sources.

We should update DSA to make use of custom data sources to use official best practices instead of the hand-rolled graphics implementation. In the end, the message feeds should be represented by DynamicEntityLayer with DynamicEntityDataSource instead of GraphicsOverlay.

All work should be done off of the v.next branch, which will build with Qt 6.5.1 and Maps SDK 200.2.

# Definition of done

- [x] App is updated to use DynamicEntityLayer for displaying real time feeds
- [x] Updates work for GeoMessage and Cursor On Target
- [ ] All other functions in the app continue to work as they did previously, such as Alerts

**Comment by kerryrobinson** [comment]
Instead of creating a new issue, I'll add a comment here on issues/bugs I've noticed with the update to use Dynamic Entities, since it is not merged yet.  We can determine if we need to turn any of these into new issues. 

1- The `Message Feeds` panel lists duplicates of the 7 feeds. The bottom set of feeds in the list do not have any effect on turning on or off the visibility of the feeds (only the top set).  I noticed that this only happens the first time I open the app and if there hasn't already been a config file created. Once the config file is created, I cannot reproduce this issue. 
![image](https://github.com/Esri/dynamic-situational-awareness-qt/assets/3383476/71c3b497-3293-431b-8a86-806e76d607c0)

2 - Dynamic Entities layers show up in the `Overlays` panel, when they should only appear in the `Message Feeds` panel. (note these are also duplicated when the app opens for the first time without an existing config file present)
![image](https://github.com/Esri/dynamic-situational-awareness-qt/assets/3383476/d47391fd-0be5-4c42-bac4-ff99f0c1fdcf)

3 - Long-pressing on a dynamic entity in a feed results in an app crash

4- The Conditions list says there are no conditions, even though there is one condition saved in the config file (the default condition to highlight any Land Friendly Tracks that have a status of 1). 

5- The `Conditions` and `Observations Report` panels UIs needs to be updated. The text on the Back and Next buttons are not visible (instead you see ellipses).  Also, hitting the X at the bottom ('Cancel')  does not cancel the panel (hitting the X in the upper right-hand corner does). 
![image](https://github.com/Esri/dynamic-situational-awareness-qt/assets/3383476/ccfba940-e582-445c-a2a1-d1f257a7d3f9)

6- Conditions no longer seem to be working at all.  The Dynamic Entity layers will need to be hooked up (they can be treated as graphics so it should technically be possible), but also `My Location` doesn't seem to be working after setting up a spatial condition.  (eventually we should support Geotriggers but for now I recommend seeing if we can hook up Dynamic Entities to work with the existing Conditions/Alerts capability in the app). 

7- Running the Line of Sight tool on the 3D observers data crashes the app.  I was only able to repro this once. 

8- Make sure to update the About dialog to reference the latest version of the SDK used and version of the app (still 2.0.0?)
